### DISCUSSION TITLE
Issue getting models through app.state

### DISCUSSION THREAD

**Discussion by gfrazzoli** [open-webui/open-webui#4820] [General]
# Bug Report

## Installation Method

docker image

## Environment

- **Open WebUI Version: 0.3.13
- **Ollama : 0.3.5

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Model fetched from app.state , in my case sentence_transformer_rf exist and is defined.

## Actual Behavior:

State has no object sentence_transformer_rf, no matter where i try to define it, either in the init, or in the function directly. 

## Description

**Bug Summary:**

I want to retrieve the app.state.sentence_transformer_rf for a function I'm building (which aim to reproduce the RAG features with permissions and stuff to sum up).
When I take a look at the backend, the reranking_function is for instance defined this way : 
 `reranking_function=app.state.sentence_transformer_rf`  (in /open-webui/tree/main/backend/apps/rag
/main.py )
and it seems to work perfectly fine, but not in my case. I do have hybrid search enabled as well. 

I tried both calling app.state.sentence_transformer_rf in def __init__ : `self.reranking_function=app.state.sentence_transformer_rf` or outside the def __init__ : `reranking_function = self.app.state.sentence_transformer_rf` ( `app = FastApi()` ;` self.app = app` defined in the __init__)

## Logs and Screenshots

**Browser Console Logs:**

Something went wrong :/ 'State' object has no attribute 'sentence_transformer_rf'

**Comment by GridexX** [comment]
I'm working with @gfrazzoli 

Please help us to solve this issue, it will help our users a lot :pray:
